lib/hostip.c

```diff
@@ -496,11 +497,15 @@ Curl_cache_addr(struct Curl_easy *data,
   entry_len = create_hostcache_id(hostname, hostlen, port,
                                   entry_id, sizeof(entry_id));
 
-  dns->inuse = 1;   /* the cache has the first reference */
+  dns->refcount = 1; /* the cache has the first reference */
   dns->addr = addr; /* this is the address(es) */
-  time(&dns->timestamp);
-  if(dns->timestamp == 0)
-    dns->timestamp = 1;   /* zero indicates permanent CURLOPT_RESOLVE entry */
+  if(permanent)
+    dns->timestamp = 0; /* an entry that never goes stale */
+  else {
+    time(&dns->timestamp);
```

Comment:
```suggestion
    dns->timestamp = time(NULL);
```

For consistency with how we use it elsewhere, but I also recall that some system somewhere caused us grief otherwise.

Reply:
Changed.